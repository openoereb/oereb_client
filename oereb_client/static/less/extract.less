/**
 * Extract styling
 */

@import "variables";

div#extract {
  width: @menu-with;
  max-width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  height: 0;
  padding-top: @header-height;
  background-color: #fff;
  transition: height @animation-duration;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: stretch;
  div.extract-wrapper {
    z-index: 1;
    background-color: @menu-background-color;
    overflow-y: scroll;
    flex: 1;
    padding-bottom: @default-padding;
    margin-top: @default-padding;
  }
  span.extract-close {
    z-index: 2;
    display: block;
    float: right;
    color: @brand-primary;
    opacity: 1;
  }
  span.extract-close:hover {
    color: #000000;
    opacity: 1;
  }
  .panel-default > .panel-heading .badge {
    background-color: @brand-primary;
  }
  .collapse-extract {
    padding-top: @default-padding;
    .btn {
      border-bottom-width: 0;
    }
  }
}

div#extract.loading {
  height: @loading-indicator-height;
}

div#extract.active {
  height: 100%;
}

div#extract.information {
  box-shadow: @menu-shadow;
}