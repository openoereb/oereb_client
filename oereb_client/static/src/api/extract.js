export function queryExtractById(applicationUrl, egrid) {
    const url = new URL(applicationUrl + 'extract/reduced/json/geometry/' + egrid);
    url.searchParams.append('_dc', new Date().getTime());
    return fetch(url).then(response => response.json());
}

export function sanitizeTopicCode(code) {
    return code.replace('.', '').replace('-', '').replace('_', '');
}

export function groupRestrictionsByTopic(retstrictions) {
    const restrictionsByTopic = {};
    retstrictions.forEach((restriction) => {
        const code = sanitizeTopicCode(restriction.Theme.Code);
        if (!restrictionsByTopic.hasOwnProperty(code)) {
            restrictionsByTopic[code] = [];
        }
        restrictionsByTopic[code].push(restriction);
    });
    return restrictionsByTopic;
}
