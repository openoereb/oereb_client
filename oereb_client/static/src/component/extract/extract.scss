@import "../../oereb_client.scss";

div.oereb-client-extract-wrapper {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    background-color: $menu-background-color;
    box-shadow: $default-shadow;
    overflow: hidden;
    transition: height $animation-duration linear;

    div.oereb-client-extract {
        width: 100%;
        height: 100%;
        padding-top: $header-height;
        overflow: hidden;
    }
    
    div.oereb-client-extract.data {
        .category:not(.show) {
            border: 0px;
        }
    
        & > div.accordion {
            overflow-y: scroll;
    
            button.accordion-button:focus,
            button.accordion-button:not(.collapsed) {
                box-shadow: none;
            }
        }

        @include media-breakpoint-up(xl) {
            button.collapse-extract {
                display: none !important;
            }
        }
    }
}

div.oereb-client-extract-wrapper.hidden {
    height: $header-height;
}

div.oereb-client-extract-wrapper.loading {
    height: $header-height + 100px;
}

div.oereb-client-extract-wrapper.shown {
    height: 100%;
}

div.oereb-client-extract-wrapper.collapsed {
    height: $header-height + 50px;
}

@include media-breakpoint-up(xl) {
    div.oereb-client-extract-wrapper {
        width: $menu-with;
    }
}