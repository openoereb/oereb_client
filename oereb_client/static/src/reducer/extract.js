import { createSlice } from "@reduxjs/toolkit";

export const extractSlice = createSlice({
    name: 'extract',
    initialState: {
        loading: false,
        visible: false,
        error: false,
        egrid: null,
        data: {}
    },
    reducers: {
        loadExtract: (state, action) => {
            console.log('Extract loading');
            if (!state.loading) {
                state.loading = true;
                state.visible = false;
                state.error = false;
                state.egrid = action.payload.egrid;
                state.data = {};
            }
        },
        showExtract: (state, action) => {
            state.loading = false;
            state.visible = true;
            state.error = false;
            state.data = action.payload.extract;
            console.log('Extract loaded', state.data);
        },
        showError: (state, action) => {
            state.loading = false;
            state.visible = false;
            state.error = true;
            state.data = {};
            console.log('Extract failed');
        },
        hideExtract: (state, action) => {
            state.loading = false;
            state.visible = false;
            state.error = false;
            state.egrid = null;
            state.data = {};
            console.log('Extract closed');
        }
    }
});

export const { loadExtract, showExtract, showError, hideExtract } = extractSlice.actions;

export default extractSlice.reducer;
