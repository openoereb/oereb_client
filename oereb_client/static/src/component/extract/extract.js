import './extract.scss';

import React from 'react';
import { useSelector } from 'react-redux';
import OerebExtractLoading from './extract_loading';
import OerebExtractError from './extract_error';
import OerebExtractData from './extract_data';

function OerebExtract(props) {
    const extract = useSelector((state) => state.extract);

    function getContent(cfg) {
        if (cfg.loading) {
            return (
                <OerebExtractLoading />
            );
        }
        else if (cfg.error) {
            return (
                <OerebExtractError />
            );
        }
        else if (cfg.visible) {
            return (
                <OerebExtractData data={extract.data.GetExtractByIdResponse.extract} />
            );
        }
        else {
            return null;
        }
    }

    const content = getContent(extract);

    return (
        <div class="oereb-client-extract">
            {content}
        </div>
    );
}

export default OerebExtract;
