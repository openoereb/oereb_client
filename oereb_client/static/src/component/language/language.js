import PropTypes from 'prop-types';
import React from "react";
import {withNamespaces} from 'react-i18next';
import {useDispatch, useSelector} from "react-redux";

import {setLanguage} from "../../reducer/language";

const OerebLanguage = function({t}) {
  const dispatch = useDispatch();
  const language = useSelector((state) => state.language);
  const currentLanguage = language.current;
  const availableLanguages = language.available;

  const selectLanguage = function(lang) {
    dispatch(setLanguage(lang));
  };

  const languages = availableLanguages.map((language, key) =>
    <li key={key}>
      <button className="dropdown-item" onClick={selectLanguage.bind(this, language)}>
        {language.toUpperCase()}
      </button>
    </li>
  );

  return (
    <div>
      <button className="btn btn-outline-secondary dropdown-toggle"
        type="button"
        data-bs-toggle="dropdown"
        aria-expanded="false"
        title={t('menu.language.title')}>
        {currentLanguage.toUpperCase()}
      </button>
      <ul className="dropdown-menu">
        {languages}
      </ul>
    </div>
  );
};

OerebLanguage.propTypes = {
  t: PropTypes.func.isRequired
};

export default withNamespaces()(OerebLanguage);
