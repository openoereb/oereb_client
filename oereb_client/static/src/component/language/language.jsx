import React from "react";
import {useTranslation} from "react-i18next";
import {useDispatch, useSelector} from "react-redux";

import {hideExtract, loadExtract, showError, showExtract} from "../../reducer/extract";
import {updateHistory} from "../../reducer/history";
import {setLanguage} from "../../reducer/language";
import {queryExtractById} from "../../request/extract";
import {TooManyRequestsError} from "../../util/error";
import {showWarning} from "../../reducer/message";

/**
 * This component provides the language selector.
 */
const OerebLanguage = function() {
  const {t} = useTranslation();
  const dispatch = useDispatch();
  const config = useSelector((state) => state.config).config;
  const serviceUrl = config.service_url;
  const language = useSelector((state) => state.language);
  const currentLanguage = language.current;
  const availableLanguages = language.available;
  const extract = useSelector((state) => state.extract);

  const selectLanguage = function(lang) {
    if (lang !== currentLanguage) {
      dispatch(setLanguage(lang));
      if (extract.visible) {
        dispatch(loadExtract({
          egrid: extract.egrid,
          identdn: extract.identdn,
          number: extract.number,
          zoom: false
        }));
        queryExtractById(
          serviceUrl,
          extract.egrid,
          extract.identdn,
          extract.number,
          config.extract_json_timeout,
          lang
        )
          .then((data) => {
            dispatch(showExtract(data));
            dispatch(updateHistory(data));
          })
          .catch((error) => {
            if (error instanceof TooManyRequestsError) {
              dispatch(hideExtract());
              dispatch(showWarning(t('extract.error.too_many_requests'), true));
            }
            else {
              dispatch(showError());
            }
          });
      }
    }
  };

  const languages = availableLanguages.map((language, key) =>
    <li key={key}>
      <button className="dropdown-item" onClick={selectLanguage.bind(this, language)}>
        {language.toUpperCase()}
      </button>
    </li>
  );

  return (
    <div className="oereb-client-language-selector">
      <button className="btn btn-outline-secondary dropdown-toggle"
        type="button"
        data-bs-toggle="dropdown"
        aria-expanded="false"
        title={t("menu.language.title")}>
        {currentLanguage.toUpperCase()}
      </button>
      <ul className="dropdown-menu">
        {languages}
      </ul>
    </div>
  );
};

export default OerebLanguage;
