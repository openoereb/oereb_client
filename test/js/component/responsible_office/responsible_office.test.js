import React from "react";
import {render} from '@testing-library/react';
import {act} from "react-dom/test-utils";
import {Provider} from "react-redux";

import OerebResponsibleOffice from
  "../../../../oereb_client/static/src/component/responsible_office/responsible_office";
import {initLanguages} from "../../../../oereb_client/static/src/reducer/language";
import {groupRestrictionsByTopic} from "../../../../oereb_client/static/src/request/extract";
import MainStore from "../../../../oereb_client/static/src/store/main";
import extract from "../../../../samples/extract.json";

describe('responsible office component', () => {

  let component;

  beforeEach(() => {
    act(() => {
      MainStore.dispatch(initLanguages({
        default: 'de',
        available: ['de']
      }));
    });
    const restrictions = groupRestrictionsByTopic(
      extract.GetExtractByIdResponse.extract.RealEstate.RestrictionOnLandownership,
      extract.GetExtractByIdResponse.extract.ConcernedTheme
    )['chStatischeWaldgrenzen']['inForce'];
    component = render(
      <Provider store={MainStore}>
        <OerebResponsibleOffice restrictions={restrictions} />
      </Provider>
    );
  });

  it('should render responsible office', () => {
    expect(component.asFragment()).toMatchSnapshot();
  });

});