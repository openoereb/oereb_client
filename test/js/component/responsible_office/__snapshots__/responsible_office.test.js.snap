// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`responsible office component should render responsible office 1`] = `
<Provider
  store={
    {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <OerebResponsibleOffice
    restrictions={
      [
        {
          "Geometry": [
            {
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "Line": {
                "coordinates": [
                  [
                    2608906.85,
                    1261915.2,
                  ],
                  [
                    2608887.35,
                    1261815.76,
                  ],
                ],
                "crs": "EPSG:2056",
                "type": "LineString",
              },
            },
          ],
          "Lawstatus": {
            "Code": "inForce",
            "Text": [
              {
                "Language": "de",
                "Text": "Rechtskräftig",
              },
            ],
          },
          "LegalProvisions": [
            {
              "Index": 2,
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "ResponsibleOffice": {
                "Name": [
                  {
                    "Language": "de",
                    "Text": "Bundesamt für Verkehr BAV",
                  },
                ],
                "OfficeAtWeb": [
                  {
                    "Language": "de",
                    "Text": "https://www.bav.admin.ch/bav/de/home.html",
                  },
                ],
              },
              "TextAtWeb": [
                {
                  "Language": "de",
                  "Text": "https://oereb-gr-preview.000.ch/api/attachments/197",
                },
              ],
              "Title": [
                {
                  "Language": "de",
                  "Text": "Baugesetz, 3891.100",
                },
              ],
              "Type": {
                "Code": "LegalProvision",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtsvorschrift",
                  },
                ],
              },
            },
            {
              "Abbreviation": [
                {
                  "Language": "de",
                  "Text": "KRVO",
                },
              ],
              "Index": 1,
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "OfficialNumber": [
                {
                  "Language": "de",
                  "Text": "BR 801.110",
                },
              ],
              "ResponsibleOffice": {
                "Name": [
                  {
                    "Language": "de",
                    "Text": "Bundesamt für Verkehr BAV",
                  },
                ],
                "OfficeAtWeb": [
                  {
                    "Language": "de",
                    "Text": "https://www.bav.admin.ch/bav/de/home.html",
                  },
                ],
              },
              "TextAtWeb": [
                {
                  "Language": "de",
                  "Text": "https://www.gr-lex.gr.ch/frontend/versions/pdf_file_with_annex/2934?locale=de",
                },
              ],
              "Title": [
                {
                  "Language": "de",
                  "Text": "Raumplanungsverordnung für den Kanton Graubünden",
                },
              ],
              "Type": {
                "Code": "Law",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Gesetzliche Grundlage",
                  },
                ],
              },
            },
            {
              "Abbreviation": [
                {
                  "Language": "de",
                  "Text": "RPG",
                },
              ],
              "Index": 1,
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "OfficialNumber": [
                {
                  "Language": "de",
                  "Text": "SR 700",
                },
              ],
              "ResponsibleOffice": {
                "Name": [
                  {
                    "Language": "de",
                    "Text": "Bundesamt für Verkehr BAV",
                  },
                ],
                "OfficeAtWeb": [
                  {
                    "Language": "de",
                    "Text": "https://www.bav.admin.ch/bav/de/home.html",
                  },
                ],
              },
              "TextAtWeb": [
                {
                  "Language": "de",
                  "Text": "http://www.lexfind.ch/dtah/167348/2",
                },
              ],
              "Title": [
                {
                  "Language": "de",
                  "Text": "Bundesgesetz über die Raumplanung (Raumplanungsgesetz, RPG)",
                },
              ],
              "Type": {
                "Code": "Law",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Gesetzliche Grundlage",
                  },
                ],
              },
            },
            {
              "Abbreviation": [
                {
                  "Language": "de",
                  "Text": "KRG",
                },
              ],
              "Index": 1,
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "OfficialNumber": [
                {
                  "Language": "de",
                  "Text": "BR 801.100",
                },
              ],
              "ResponsibleOffice": {
                "Name": [
                  {
                    "Language": "de",
                    "Text": "Bundesamt für Verkehr BAV",
                  },
                ],
                "OfficeAtWeb": [
                  {
                    "Language": "de",
                    "Text": "https://www.bav.admin.ch/bav/de/home.html",
                  },
                ],
              },
              "TextAtWeb": [
                {
                  "Language": "de",
                  "Text": "https://www.gr-lex.gr.ch/frontend/versions/pdf_file_with_annex/2936?locale=de",
                },
              ],
              "Title": [
                {
                  "Language": "de",
                  "Text": "Raumplanungsgesetz für den Kanton Graubünden",
                },
              ],
              "Type": {
                "Code": "Law",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Gesetzliche Grundlage",
                  },
                ],
              },
            },
            {
              "Index": 2,
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "ResponsibleOffice": {
                "Name": [
                  {
                    "Language": "de",
                    "Text": "Bundesamt für Verkehr BAV",
                  },
                ],
                "OfficeAtWeb": [
                  {
                    "Language": "de",
                    "Text": "https://www.bav.admin.ch/bav/de/home.html",
                  },
                ],
              },
              "TextAtWeb": [
                {
                  "Language": "de",
                  "Text": "https://oereb-gr-preview.000.ch/api/attachments/213",
                },
              ],
              "Title": [
                {
                  "Language": "de",
                  "Text": "Revision Ortsplanung, 07.447",
                },
              ],
              "Type": {
                "Code": "LegalProvision",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtsvorschrift",
                  },
                ],
              },
            },
            {
              "Index": 2,
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "ResponsibleOffice": {
                "Name": [
                  {
                    "Language": "de",
                    "Text": "Bundesamt für Verkehr BAV",
                  },
                ],
                "OfficeAtWeb": [
                  {
                    "Language": "de",
                    "Text": "https://www.bav.admin.ch/bav/de/home.html",
                  },
                ],
              },
              "TextAtWeb": [
                {
                  "Language": "de",
                  "Text": "https://oereb-gr-preview.000.ch/api/attachments/198",
                },
              ],
              "Title": [
                {
                  "Language": "de",
                  "Text": "Baugesetz, 3891.100",
                },
              ],
              "Type": {
                "Code": "LegalProvision",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtsvorschrift",
                  },
                ],
              },
            },
            {
              "Index": 2,
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "ResponsibleOffice": {
                "Name": [
                  {
                    "Language": "de",
                    "Text": "Bundesamt für Verkehr BAV",
                  },
                ],
                "OfficeAtWeb": [
                  {
                    "Language": "de",
                    "Text": "https://www.bav.admin.ch/bav/de/home.html",
                  },
                ],
              },
              "TextAtWeb": [
                {
                  "Language": "de",
                  "Text": "https://oereb-gr-preview.000.ch/api/attachments/214",
                },
              ],
              "Title": [
                {
                  "Language": "de",
                  "Text": "Revision Ortsplanung, 07.447",
                },
              ],
              "Type": {
                "Code": "LegalProvision",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtsvorschrift",
                  },
                ],
              },
            },
            {
              "Index": 3,
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "ResponsibleOffice": {
                "Name": [
                  {
                    "Language": "de",
                    "Text": "Bundesamt für Verkehr BAV",
                  },
                ],
                "OfficeAtWeb": [
                  {
                    "Language": "de",
                    "Text": "https://www.bav.admin.ch/bav/de/home.html",
                  },
                ],
              },
              "TextAtWeb": [
                {
                  "Language": "de",
                  "Text": "http://example.com/goodhint1",
                },
              ],
              "Title": [
                {
                  "Language": "de",
                  "Text": "Guter Ratschlag",
                },
              ],
              "Type": {
                "Code": "Hint",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Hinweis",
                  },
                ],
              },
            },
            {
              "Index": 3,
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "ResponsibleOffice": {
                "Name": [
                  {
                    "Language": "de",
                    "Text": "Bundesamt für Verkehr BAV",
                  },
                ],
                "OfficeAtWeb": [
                  {
                    "Language": "de",
                    "Text": "https://www.bav.admin.ch/bav/de/home.html",
                  },
                ],
              },
              "TextAtWeb": [
                {
                  "Language": "de",
                  "Text": "http://example.com/goodhint2",
                },
              ],
              "Title": [
                {
                  "Language": "de",
                  "Text": "Guter Ratschlag",
                },
              ],
              "Type": {
                "Code": "Hint",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Hinweis",
                  },
                ],
              },
            },
            {
              "Index": 3,
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "ResponsibleOffice": {
                "Name": [
                  {
                    "Language": "de",
                    "Text": "Bundesamt für Verkehr BAV",
                  },
                ],
                "OfficeAtWeb": [
                  {
                    "Language": "de",
                    "Text": "https://www.bav.admin.ch/bav/de/home.html",
                  },
                ],
              },
              "TextAtWeb": [
                {
                  "Language": "de",
                  "Text": "http://example.com/goodhint3",
                },
              ],
              "Title": [
                {
                  "Language": "de",
                  "Text": "Seht guter Ratschlag mit langem titel: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras convallis venenatis posuere. Nunc sollicitudin fermentum bibendum. Nulla ut massa enim. Sed rhoncus euismod venenatis",
                },
              ],
              "Type": {
                "Code": "Hint",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Hinweis",
                  },
                ],
              },
            },
          ],
          "LegendText": [
            {
              "Language": "de",
              "Text": "Belastet, untersuchungsbedürftig",
            },
          ],
          "LengthShare": 101,
          "Map": {
            "ReferenceWMS": [
              {
                "Language": "de",
                "Text": "https://wms.geo.admin.ch/?SERVICE=WMS&REQUEST=GetMap&VERSION=1.3.0&STYLES=default&CRS=EPSG%3A2056&BBOX=2608564.1173571427%2C1261670.8114999998%2C2609194.3826428573%2C1262028.7715&WIDTH=493&HEIGHT=280&FORMAT=image%2Fpng&LAYERS=ch.bav.kataster-belasteter-standorte-oev.oereb&SRS=EPSG%3A2056",
              },
            ],
            "layerIndex": 1,
            "layerOpacity": 0.75,
            "max": {
              "coordinates": [
                2609194.3826428573,
                1262028.7715,
              ],
              "crs": "EPSG:2056",
              "type": "Point",
            },
            "min": {
              "coordinates": [
                2608564.1173571427,
                1261670.8114999998,
              ],
              "crs": "EPSG:2056",
              "type": "Point",
            },
          },
          "ResponsibleOffice": {
            "Name": [
              {
                "Language": "de",
                "Text": "Bundesamt für Verkehr BAV",
              },
            ],
            "OfficeAtWeb": [
              {
                "Language": "de",
                "Text": "https://www.bav.admin.ch/bav/de/home.html",
              },
            ],
          },
          "SymbolRef": "samples/static/symbol2.png",
          "Theme": {
            "Code": "ch.StatischeWaldgrenzen",
            "Text": [
              {
                "Language": "de",
                "Text": "Statische Waldgrenzen",
              },
            ],
          },
          "TypeCode": "StaoTyp2",
          "TypeCodelist": "https://models.geo.admin.ch/BAFU/KbS_Codetexte_V1_4.xml",
        },
        {
          "Geometry": [
            {
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "Line": {
                "coordinates": [
                  [
                    2608906.85,
                    1261915.2,
                  ],
                  [
                    2608887.35,
                    1261815.76,
                  ],
                ],
                "crs": "EPSG:2056",
                "type": "LineString",
              },
            },
          ],
          "Lawstatus": {
            "Code": "inForce",
            "Text": [
              {
                "Language": "de",
                "Text": "Rechtskräftig",
              },
            ],
          },
          "LegalProvisions": [
            {
              "Index": 2,
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "ResponsibleOffice": {
                "Name": [
                  {
                    "Language": "de",
                    "Text": "Bundesamt für Verkehr BAV",
                  },
                ],
                "OfficeAtWeb": [
                  {
                    "Language": "de",
                    "Text": "https://www.bav.admin.ch/bav/de/home.html",
                  },
                ],
              },
              "TextAtWeb": [
                {
                  "Language": "de",
                  "Text": "https://oereb-gr-preview.000.ch/api/attachments/197",
                },
              ],
              "Title": [
                {
                  "Language": "de",
                  "Text": "Baugesetz, 3891.100",
                },
              ],
              "Type": {
                "Code": "LegalProvision",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtsvorschrift",
                  },
                ],
              },
            },
            {
              "Abbreviation": [
                {
                  "Language": "de",
                  "Text": "KRVO",
                },
              ],
              "Index": 1,
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "OfficialNumber": [
                {
                  "Language": "de",
                  "Text": "BR 801.110",
                },
              ],
              "ResponsibleOffice": {
                "Name": [
                  {
                    "Language": "de",
                    "Text": "Bundesamt für Verkehr BAV",
                  },
                ],
                "OfficeAtWeb": [
                  {
                    "Language": "de",
                    "Text": "https://www.bav.admin.ch/bav/de/home.html",
                  },
                ],
              },
              "TextAtWeb": [
                {
                  "Language": "de",
                  "Text": "https://www.gr-lex.gr.ch/frontend/versions/pdf_file_with_annex/2934?locale=de",
                },
              ],
              "Title": [
                {
                  "Language": "de",
                  "Text": "Raumplanungsverordnung für den Kanton Graubünden",
                },
              ],
              "Type": {
                "Code": "Law",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Gesetzliche Grundlage",
                  },
                ],
              },
            },
            {
              "Abbreviation": [
                {
                  "Language": "de",
                  "Text": "RPG",
                },
              ],
              "Index": 1,
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "OfficialNumber": [
                {
                  "Language": "de",
                  "Text": "SR 700",
                },
              ],
              "ResponsibleOffice": {
                "Name": [
                  {
                    "Language": "de",
                    "Text": "Bundesamt für Verkehr BAV",
                  },
                ],
                "OfficeAtWeb": [
                  {
                    "Language": "de",
                    "Text": "https://www.bav.admin.ch/bav/de/home.html",
                  },
                ],
              },
              "TextAtWeb": [
                {
                  "Language": "de",
                  "Text": "http://www.lexfind.ch/dtah/167348/2",
                },
              ],
              "Title": [
                {
                  "Language": "de",
                  "Text": "Bundesgesetz über die Raumplanung (Raumplanungsgesetz, RPG)",
                },
              ],
              "Type": {
                "Code": "Law",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Gesetzliche Grundlage",
                  },
                ],
              },
            },
            {
              "Abbreviation": [
                {
                  "Language": "de",
                  "Text": "KRG",
                },
              ],
              "Index": 1,
              "Lawstatus": {
                "Code": "inForce",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Rechtskräftig",
                  },
                ],
              },
              "OfficialNumber": [
                {
                  "Language": "de",
                  "Text": "BR 801.100",
                },
              ],
              "ResponsibleOffice": {
                "Name": [
                  {
                    "Language": "de",
                    "Text": "Bundesamt für Verkehr BAV",
                  },
                ],
                "OfficeAtWeb": [
                  {
                    "Language": "de",
                    "Text": "https://www.bav.admin.ch/bav/de/home.html",
                  },
                ],
              },
              "TextAtWeb": [
                {
                  "Language": "de",
                  "Text": "https://www.gr-lex.gr.ch/frontend/versions/pdf_file_with_annex/2936?locale=de",
                },
              ],
              "Title": [
                {
                  "Language": "de",
                  "Text": "Raumplanungsgesetz für den Kanton Graubünden",
                },
              ],
              "Type": {
                "Code": "Law",
                "Text": [
                  {
                    "Language": "de",
                    "Text": "Gesetzliche Grundlage",
                  },
                ],
              },
            },
          ],
          "LegendText": [
            {
              "Language": "de",
              "Text": "Belastet, untersuchungsbedürftig",
            },
          ],
          "LengthShare": 101,
          "Map": {
            "ReferenceWMS": [
              {
                "Language": "de",
                "Text": "https://wms.geo.admin.ch/?SERVICE=WMS&REQUEST=GetMap&VERSION=1.3.0&STYLES=default&CRS=EPSG%3A2056&BBOX=2608564.1173571427%2C1261670.8114999998%2C2609194.3826428573%2C1262028.7715&WIDTH=493&HEIGHT=280&FORMAT=image%2Fpng&LAYERS=ch.bav.kataster-belasteter-standorte-oev.oereb&SRS=EPSG%3A2056",
              },
            ],
            "layerIndex": 1,
            "layerOpacity": 0.75,
            "max": {
              "coordinates": [
                2609194.3826428573,
                1262028.7715,
              ],
              "crs": "EPSG:2056",
              "type": "Point",
            },
            "min": {
              "coordinates": [
                2608564.1173571427,
                1261670.8114999998,
              ],
              "crs": "EPSG:2056",
              "type": "Point",
            },
          },
          "ResponsibleOffice": {
            "Name": [
              {
                "Language": "de",
                "Text": "Bundesamt für Verkehr BAV",
              },
            ],
            "OfficeAtWeb": [
              {
                "Language": "de",
                "Text": "https://www.bav.admin.ch/bav/de/home.html",
              },
            ],
          },
          "SymbolRef": "samples/static/symbol2.png",
          "Theme": {
            "Code": "ch.StatischeWaldgrenzen",
            "Text": [
              {
                "Language": "de",
                "Text": "Statische Waldgrenzen",
              },
            ],
          },
          "TypeCode": "StaoTyp2",
          "TypeCodelist": "https://models.geo.admin.ch/BAFU/KbS_Codetexte_V1_4.xml",
        },
      ]
    }
  >
    <dl
      className="oereb-client-responsible-office border-top pt-2 mb-2"
    >
      <dt>
        extract.topic.responsible_office
      </dt>
      <dd
        className="ms-2"
        key="0"
      >
        <a
          href="https://www.bav.admin.ch/bav/de/home.html"
          rel="noreferrer"
          target="_blank"
        >
          Bundesamt für Verkehr BAV
        </a>
      </dd>
    </dl>
  </OerebResponsibleOffice>
</Provider>
`;
